### ISSUE TITLE
command not found: tmux_detach

### ISSUE THREAD

**Issue by mbutz** [Vifon/zranger#1]
Hi @Vifon,

I followed [your advice](https://github.com/ranger/ranger/issues/704) and thereby tried `zsh`. I think I'll stick with it. Thanks a lot for this hint.

I followed also your installation notes, checked it twice but switching to `ranger` gives me a:

```
command not found: 'tmux_detach'
```
I have

* the snippet loading `zranger` in my `zshrc`; I also set the `fpath` variable
* `zranger` is in `~/.fpath`
* I copied `zranger_chdir_handler ...` and `class tmux_detach ...` to rangers `commands.py`
* I defined a keybinding in my `rc.conf`
* I also checked and fullfilled the dependencies (added `tmux`)

I probably missed something but can*t find out what.

Martin

**Comment by Vifon** [comment]
Are you sure you have copied properly the `class tmux_detach(Command): ...` snippet? It seems ranger cannot find this command. Check once again.

Since this command is optional, try detaching from tmux manually, by pressing <kbd>ctrl+b</kbd> and then <kbd>d</kbd>. Is it working?

**Comment by mbutz** [comment]
Yes, checked again. Snippets are in place. First thing in my `commands.py`. I know, obviously `ranger` can't find the command. I did copy the whole `ranger` config which no resides under `~/.config/ranger`. I am pretty sure it works because my personal keybindings in `rc.config` do also work.

Tmux and detaching works in `zsh`:

```
$ ~ % tmux
$ [detached (from session 0)]
```

Can it have something to do with the order of my `commands.py`? I have your two snippets on the top of that file (which I initially copied from default configuration):

```
# comming from copied default commands.py
from ranger.api.commands import * 

# snippets from zranger
import signal
def zranger_chdir_handler(signal, frame):
    with open("/tmp/zranger-cwd", "r") as f:
        Command.fm.cd(f.readline().strip())
        os.unlink("/tmp/zranger-cwd")
        signal.signal(signal.SIGUSR1, zranger_chdir_handler)

        class tmux_detach(Command):
            """
            :tmux_detach

            Detach from this tmux session (if inside tmux).
            """
            def execute(self):
                if not os.environ.get('TMUX'):
                    return
            os.system("tmux detach")

# rest of the code comming from copied default commands.py
```

Thanks
Martin

**Comment by Vifon** [comment]
I see the problem now. :)

Python is sensitive to indentation. Your `tmux_detach` class is inside the `zranger_chdir_handler` function. You need to paste the code exactly as it was in the readme.

**Comment by mbutz** [comment]
Yes. I did copy and paste the code exactly but spacemacs messed it up (probably because I hadn't installed the python layer). I was already wondering why the pasted part had a different indentation then the rest of my `commands.py` but I did not really assume that this is crucial.

Additionally I did not start with your keybindings. I then realized that I'd have to adjust the bindings in two places (`.zshrc` and `rc.conf`) so now it works!

Brilliant. Does exactly what I want. Thanks a lot!

Martin
